### PR TITLE
SDIT-3835 Handle court schedule event edge cases

### PR DESCRIPTION
**Pull request by mikehalmamoj**

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/movements/court/CourtSchedulerScheduleIntTest.kt [hunk 1]
```diff
@@ -109,6 +114,182 @@ class CourtSchedulerScheduleIntTest(
           )
         }
       }
+
+      @Nested
+      inner class HappyPathUpdated {
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/movements/court/CourtSchedulerScheduleIntTest.kt:119]: You can see what's been added to the court schedule sync from the tests: update existing, mapping failures, duplicate mappings, created in NOMIS.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/integration/IntegrationTestBase.kt [hunk 2]
```diff
@@ -71,7 +71,7 @@ import uk.gov.justice.hmpps.test.kotlin.auth.JwtAuthorisationHelper
 @ExtendWith(OutputCaptureExtension::class)
 abstract class IntegrationTestBase {
 
-  @MockitoSpyBean
+  @MockitoBean
   lateinit var telemetryClient: TelemetryClient
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/integration/IntegrationTestBase.kt:75]: This stops the VPN adding 5 seconds to tests at random. Stupid VPN.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/movements/court/CourtSchedulerAppearanceService.kt [hunk 3]
```diff
@@ -29,27 +37,75 @@ class CourtSchedulerAppearanceService(
...
-    val nomis = nomisApi.upsertCourtScheduleOut(prisonerNumber, dps.toNomisUpsertRequest())
-      .also {
-        telemetryMap["bookingId"] = it.bookingId.toString()
-        telemetryMap["nomisEventId"] = it.eventId.toString()
+    if (event.additionalInformation.source != "DPS") {
+      telemetryClient.trackEvent("$telemetryKey-ignored", telemetryMap)
+      return
+    }
+
+    runCatching {
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/movements/court/CourtSchedulerAppearanceService.kt:47]: don't we have a `track` type function in this project?

**Review by andymarke — APPROVED** [review]
